### PR TITLE
solution

### PR DESCRIPTION
**Pull request by AlexanderTsvirko**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,5 +1,69 @@
...
+
 # write your code here
+class BaseRobot:
+    def __init__(self, name: str, weight: int, coords: list = None) -> None:
+        self.name = name
+        self.weight = weight
+        if coords is None:
+            self.coords = [0, 0]
+        else:
+            self.coords = coords
```
Comment (OleksandrPozden) [inline app/main.py:17]: write it in one line

File: app/main.py [hunk 2]
```diff
@@ -1,5 +1,69 @@
...
+        return f"Robot: {self.name}, Weight: {self.weight}"
+
+
+class FlyingRobot(BaseRobot):
+    def __init__(self, name: str, weight: int, coords: list = None) -> None:
+        super().__init__(name, weight, coords)
+        if coords is None:
+            self.coords = [0, 0, 0]
+        else:
+            self.coords = coords
```
Comment (OleksandrPozden) [inline app/main.py:41]: in one line

**Review by OleksandrPozden — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 1 above)
Comment (AlexanderTsvirko) [reply app/main.py]: Done

File: app/main.py [hunk 3]
```diff
@@ -1,5 +1,63 @@
...
+    def __init__(self, weight: int) -> None:
         self.weight = weight
 
+
 # write your code here
+class BaseRobot:
+    def __init__(self, name: str, weight: int, coords: list = None) -> None:
+        self.name = name
+        self.weight = weight
+        self.coords = [0, 0] if coords is None else coords
```
Comment (OleksandrPozden) [inline app/main.py:14]: Good, but you do it even more simple

**Review by OleksandrPozden — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 4]
```diff
@@ -1,5 +1,63 @@
...
+    def go_left(self, step: int = 1) -> None:
+        self.coords[0] -= step
+
+    def get_info(self) -> str:
+        return f"Robot: {self.name}, Weight: {self.weight}"
+
+
+class FlyingRobot(BaseRobot):
+    def __init__(self, name: str, weight: int, coords: list = None) -> None:
+        super().__init__(name, weight, coords)
```
Comment (Serhii-Leonenko) [inline app/main.py:34]:
```suggestion
        super().__init__(name, weight)
```

**Review by Serhii-Leonenko — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 5]
```diff
@@ -1,5 +1,63 @@
...
+    def __init__(self, weight: int) -> None:
         self.weight = weight
 
+
 # write your code here
+class BaseRobot:
+    def __init__(self, name: str, weight: int, coords: list = None) -> None:
+        self.name = name
+        self.weight = weight
+        self.coords = coords if coords else [0, 0]
```
Comment (OleksandrPozden) [inline app/main.py:14]: yes, you have done a great job. but look at this simple solution
```suggestion
        self.coords = coords or [0, 0]
```

File: app/main.py [hunk 6]
```diff
@@ -1,5 +1,63 @@
...
+        self.coords[0] -= step
+
+    def get_info(self) -> str:
+        return f"Robot: {self.name}, Weight: {self.weight}"
+
+
+class FlyingRobot(BaseRobot):
+    def __init__(self, name: str, weight: int, coords: list = None) -> None:
+        super().__init__(name, weight)
+        self.coords = coords if coords else [0, 0, 0]
```
Comment (OleksandrPozden) [inline app/main.py:35]: change it here, as well

**Review by OleksandrPozden — CHANGES_REQUESTED** [review]

**Review by YuriiKindrat — APPROVED** [review]

**Pull request closed without merging**